utilFunctions/getItemId.js

```diff
@@ -3,10 +3,12 @@ import extractItemId from "./extractItemId";
 // Given source (i.e. JSON response from PSS API)
 // return item's DPLA ID
 const getItemId = source =>
-  extractItemId(
-    source.mainEntity[0]["dct:references"].filter(
-      ref => ref["@type"] === "ore:Aggregation"
-    )[0]["@id"]
-  );
+  source.mainEntity[0]["dct:references"]
+    ? extractItemId(
+        source.mainEntity[0]["dct:references"].filter(
+          ref => ref["@type"] === "ore:Aggregation"
+        )[0]["@id"]
+      )
+    : null;
```

Comment:
See above comment about try-fail block - this might apply here, too.

Reply:
added